scripts/doc_checker.py

```diff
@@ -45,6 +45,8 @@
         )
 if not hasattr(sys.modules["cudf"], "DataFrame"):
     sys.modules["cudf"].DataFrame = type("DataFrame", (object,), {})
+if not hasattr(sys.modules["cupy"], "ndarray"):
+    sys.modules["cupy"].ndarray = type("ndarray", (object,), {})
```

Comment:
the `dask` import flow has suddenly [stopped working](https://github.com/dchigarev/modin/runs/5005582579?check_suite_focus=true#step:9:56) in the doc-checker script, we may also want to check whether it causes a problem in a regular `import modin.pandas as pd` import flow with the dask engine in a clean env, as a part of a separate issue ofc.

Reply:
I think we can continue this discussion/experiments in #4116, I won't close this issue after merging the PR